vignettes/dark_ink_transition/lib/dark_ink_bar.dart

```diff
@@ -3,40 +3,38 @@ import 'package:flutter/material.dart';
 import 'main.dart';
 
 class DarkInkBar extends StatefulWidget {
+  final ValueNotifier<bool>? darkModeValue;
 
-  final ValueNotifier<bool> darkModeValue;
-
-  DarkInkBar({ this.darkModeValue });
+  DarkInkBar({this.darkModeValue});
 
   @override
   State createState() {
-    return _DarkInkBarState(darkModeValue);
+    return _DarkInkBarState(darkModeValue!);
   }
-
 }
 
 class _DarkInkBarState extends State<DarkInkBar>
     with SingleTickerProviderStateMixin {
-
   static final Color darkColor = Color(0xFF171137);
   static final Color lightColor = Color(0xFF67ECDC);
 
   ValueNotifier<bool> _darkModeValue;
 
-  AnimationController _controller;
-  Animation<double> _iconOpacityAnimation;
-  Animation<double> _backgroundColorAnimation;
-  Animation<double> _foregroundColorAnimation;
+  late AnimationController _controller;
+  late Animation<double> _iconOpacityAnimation;
+  late Animation<double> _backgroundColorAnimation;
+  late Animation<double> _foregroundColorAnimation;
 
-  ImageProvider _darkModeToggleIconImage;
+  late ImageProvider _darkModeToggleIconImage;
```

Comment:
Would prefer to see these instantiated up here rather than `initState` so there is no chance of runtime error.